File: problems/Python/78. Subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []
        path = []

        def bt(start):
            res.append(list(path))
            for i in range(start, len(nums)):
                path.append(nums[i])
                bt(i + 1)
                path.pop()

        bt(0)
        return res

File: problems/C++/78. Subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> res;
        vector<int> path;
        bt(0, path, nums, res);
        return res;
    }

private:
    void bt(int start, vector<int>& path, vector<int>& nums, vector<vector<int>>& res) {
        res.push_back(path);
        for (int i = start; i < nums.size(); ++i) {
            path.push_back(nums[i]);
            bt(i + 1, path, nums, res);
            path.pop_back();
        }
    }
};